src/calendar.scss

```diff
@@ -0,0 +1,57 @@
+@import "./variables/variables.scss";
+
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  gap: $block-gap;
+  width: $calendar-width;
+  margin: auto;
+  padding-left: 0;
+  padding-top: 30px;
+  list-style: none;
+  position: relative;
+
+  @for $index from 28 through $day-count {
+    &-month-length--#{$index} {
+      & :nth-child(n + #{$index + 1}) {
+        display: none;
+      }
+    }
+  }
+
+  @each $name, $number in $days-of-week {
+    &-start-day--#{$name} {
+      & :first-child {
+        margin-left: getMargin($number, $block-size, $block-gap);
+      }
+    }
+  }
+
+  &__day {
+    @include square($block-size);
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    border: 1px solid #000;
+    background-color: #eee;
+    transition: all $transition-time;
+
+    &:hover {
+      background-color: pink;
+      cursor: pointer;
+      transform: translateY(-20px);
+    }
+
+    @for $index from 1 through $day-count {
+      &:nth-child(#{$index}) {
+        &::before {
+          content: "#{$index}";
+        }
+      }
+    }
+  }
+}
+
+@function getMargin($index, $base, $size) {
+  @return $index * ($base + $size);
```

Comment:
This modifier still not works. Try this fix, and make sure you redeploy the demo
![image](https://user-images.githubusercontent.com/118751034/218476962-b15bc5dd-7480-4a84-8e7e-02e888d61923.png)

```suggestion
  @return $index * ($base + $size) * 1px;
```

Reply:
Thanks, I have tried. hopefully it will work

Reply:
you can if it works before requesting. if you have questions, please, write in the chat